### ISSUE TITLE
Docker Error: '/opt/mfem-env/.spack-env/view/': No such file or directory when using docker

### ISSUE THREAD

**Issue by ojipadeson** [mfem/mfem#3743]
I ran docker following the command in ```config/docker/README.md```, as follows:
```
$ docker build -f config/docker/Dockerfile.base -t ghcr.io/mfem/mfem-ubuntu-base .
$ docker run -it ghcr.io/mfem/mfem-ubuntu-base bash
```
But after entering docker, the situation does not match the README:
```
$ env | grep mfem
>>> OLDPWD=/opt/mfem-env
```
But the output should actually be:
```
PKG_CONFIG_PATH=/opt/mfem-env/.spack-env/view/lib/pkgconfig:/opt/mfem-env/.spack-env/view/share/pkgconfig:/opt/mfem-env/.spack-env/view/lib64/pkgconfig
PWD=/opt/mfem-env
MANPATH=/opt/mfem-env/.spack-env/view/share/man:/opt/mfem-env/.spack-env/view/man:
CMAKE_PREFIX_PATH=/opt/mfem-env/.spack-env/view
SPACK_ENV=/opt/mfem-env
ACLOCAL_PATH=/opt/mfem-env/.spack-env/view/share/aclocal
LD_LIBRARY_PATH=/opt/mfem-env/.spack-env/view/lib:/opt/mfem-env/.spack-env/view/lib64
PATH=/opt/mfem-env/.spack-env/view/bin:/opt/view/bin:/opt/spack/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```
When I tried to run some installed miniapps, I found that there was no corresponding folder:
```
$ ls /opt/mfem-env/.spack-env/view/
>>> ls: cannot access '/opt/mfem-env/.spack-env/view/': No such file or directory
```

**Comment by jandrej** [comment]
I'd advise to use this image
```
docker pull ghcr.io/mfem/containers/developer:latest
```

**Comment by v-dobrev** [comment]
It's probably a good idea to fix this issue in `config/docker/README.md`. It maybe due to updates in the starting Docker image used by `Dockerfile.base`.

@vsoch, do you have any suggestions how to fix the issue described above?

**Comment by vsoch** [comment]
It’s late and I am reading on my phone, but the base image needs to build first, and then the image on top of it, and that’s the container being shelled into (not the base). I’ll take a closer look tomorrow when I’m more awake!

**Comment by vsoch** [comment]
Forgot to follow up here! I agree with my previously sleepy self - the issue here is shelling into the base container and expecting the instruction for the subsequent container to be the same. I'd recommend the second (multi-stage) build, however to use the base container as you are doing, you need to use the spack view where it is located (/opt/view) and then handle the environment variables on your own. Hope that helps!

**Comment by tzanio** [comment]
@ojipadeson -- does the above ☝️ suggestion, or the [Docker image](https://github.com/mfem/containers/pkgs/container/containers%2Fdeveloper) from [mfem/containers](https://github.com/mfem/containers) work for you?

**Comment by v-dobrev** [comment]
I think this maybe an issue with the README: to me it looks like @ojipadeson is following precisely the steps from this section: https://github.com/mfem/mfem/blob/master/config/docker/README.md#shell-ubuntu-base.

**Comment by vsoch** [comment]
This is the correct section to follow: https://github.com/mfem/mfem/blob/master/config/docker/README.md#shell-ubuntu. The base image section should probably be removed or updated - the intended use case is to use the multi-stage build.

**Comment by ojipadeson** [comment]
For the first method, I used the new docker provided by @jandrej ：
```
docker build -f config/docker/Dockerfile -t ghcr.io/mfem/containers/developer:latest .
docker run -it ghcr.io/mfem/containers/developer:latest
```
And get the result：
```
root@cc56bbb1b2e3:/opt/mfem-view# env | grep mfem
PWD=/opt/mfem-view
LD_LIBRARY_PATH=:/opt/mfem-view/lib:/opt/mfem-view/lib64
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/mfem-view/bin
```
I still can't find the miniapp folder.

Second, I use this multi-stage build as described by @vsoch , but I still can't find the folder:
```
docker build -f config/docker/Dockerfile -t ghcr.io/mfem/mfem-ubuntu .
docker run -it ghcr.io/mfem/mfem-ubuntu
```
Result:
```
root@14003735db48:/opt/mfem-view# env | grep mfem
PWD=/opt/mfem-view
LD_LIBRARY_PATH=:/opt/mfem-view/lib:/opt/mfem-view/lib64
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/mfem-view/bin
```

**Comment by vsoch** [comment]
@ojipadeson I can help you (and update the documentation there) in July - I just can't now because my internet data is almost maxed out, so I'm avoiding pulling / pushing containers for the rest of the month! But I'll mark my calendar for July 1st to remind myself about this first thing.

**Comment by ojipadeson** [comment]
> @ojipadeson I can help you (and update the documentation there) in July - I just can't now because my internet data is almost maxed out, so I'm avoiding pulling / pushing containers for the rest of the month and watching YouTube on low-res! 😆 😭 But I'll mark my calendar for July 1st to remind myself about this first thing.

Thank you very much and hope to get your help soon

**Comment by vsoch** [comment]
Hey folks - happy July! And hooray for me having internet bandwidth again! Watching YouTube videos on potato poly is :sob: 

As promised, here is a pull request to update the Dockerfiles and instructions: https://github.com/mfem/mfem/pull/3769.

There were a few issues - first that the main logic of the base Dockerfile.base was commented out (this was a mistake likely on my part that I missed before we merged) and I don't think I did a good job of presenting the builds in the context of use cases, e.g.,:

- "I want to develop mfem itself, and I can use spack" (base)
- "I want to use a built mfem for my own app (non-base, multi-stage slimmer build)

Please check out the PR, and instructions, and let's follow up with discussion there.

**Comment by tzanio** [comment]
Thanks @vsoch !

**Comment by vsoch** [comment]
I think this will be closed with https://github.com/mfem/mfem/pull/3769 stalebot - anything else I need to do?

**Comment by v-dobrev** [comment]
> I think this will be closed with #3769 stalebot - anything else I need to do?

No need to do anything else. Thank you for the fix -- it will be merged soon.
